### ISSUE TITLE
Django 1.10 throws admin.E013 error

### ISSUE THREAD

**Issue by nshafer** [alex/django-taggit#430]
With Django 1.10 comes a slight change to how `ModelAdmin`s are validated that will cause an `admin.E013` error if you attempt to include a TaggableManager field, such as 'tags' in a `fields` or `fieldsets` in your admin.

The error will say something like, "(admin.E013) The value of 'fieldsets[0][1]["fields"]' cannot include the many-to-many field 'tags' because that field manually specifies a relationship model."

This is because, in [the admin system check code](https://github.com/django/django/blob/master/django/contrib/admin/checks.py#L220) the check used to ignore django-taggit's TaggableManager because TaggableManager wasn't a subclass of ManyToManyField, but [now it checks](https://github.com/django/django/commit/983c158da7723eb00a376bd31db76709da4d0260#diff-044a46e68cb93c11c2dc9a7392119915R221) if `many_to_many` is True, which it is for TaggableManager, so this check now pays attention to django-taggit. And now that's it's paying attention to django-taggit, it will prevent Django from starting since our field is defining a custom through relationship, even though there is nothing special about it and it works fine as long as the check passes.

I found a workaround with is to add the `auto_created = True` Meta tag to your [custom through class](https://django-taggit.readthedocs.io/en/latest/custom_tagging.html#custom-foreignkeys):

```
class TaggedFood(TaggedItemBase):
    content_object = models.ForeignKey('Food')

    class Meta:
         auto_created = True
```

Or to TaggedItemBase in django-taggit:

```
class TaggedItemBase(ItemBase):
    class Meta:
        abstract = True
        auto_created = True
```

However, this seems like a horrible hack just to pass the system check. Any thoughts on how to fix this properly? What all does the `many_to_many = True` setting do?

**Comment by jonathanmorgan** [comment]
I'm not using a "custom through class", I think, so I'd have to alter the actual source if I wanted to triage this?  Glad I'm playing with django 1.10 on a virtual machine rather than my research server...  Let me know if I can help.

**Comment by aaugustin** [comment]
I filed an issue against Django because I think that the E013 check isn't implemented properly.

https://code.djangoproject.com/ticket/26998

For now, I'm working around this issue as follows:

{{{
class TaggedPost(TaggedItemBase):
    content_object = models.ForeignKey("blog.Post")

    # Work around https://code.djangoproject.com/ticket/26998
    class Meta:
        auto_created = True
}}}

**Comment by nshafer** [comment]
I've run into problems with my workaround preventing migrations from being made, and the ticket in Django was closed as a duplicate of a 7 year old ticket, so that's most likely not being addressed any time soon. So anyone have any ideas on anything we can do on this side for a fix?

**Comment by jonathanmorgan** [comment]
If I can get some guidance on what needs to happen, I'd be happy to do some coding.  I'd rather that than have to find a new tagging solution.  If the need is to figure out options, I'd be willing to try that, too, with some guidance on where to start.

**Comment by collinanderson** [comment]
I just added my thoughts to aaugustin's ticket. https://code.djangoproject.com/ticket/26998#comment:9

**Comment by collinanderson** [comment]
Hi All,

Here's a pull request that I think should fix it. Could you try out my patch? Leave a comment on the Pull request with your feedback.

​https://github.com/django/django/pull/7140

Thanks!
Collin

**Comment by nshafer** [comment]
Thanks, Collin, just did. For anyone else on this ticket, this was fixed by Collin and theres a PR request for integration into Django. So for now, no changes needed for django-taggit. No idea on when the patch will make it to a django release. I'll leave this ticket open until that happens for anyone else stumbling on this problem with Django 1.10. Thanks all!

**Comment by aaugustin** [comment]
Since Django started its fellowship (hi @timgraham!), we've had monthly releases (modulo security releases), so Django 1.10.1 should happen at the beginning of September.

**Comment by jonathanmorgan** [comment]
I tested the django patch and it resolved the errors for me.  Looking forward to that release.  =)

**Comment by collinanderson** [comment]
It's now been merged into django master and the 1.10 branch, so roughly 10 days from now it should get released.

**Comment by aaugustin** [comment]
Django 1.10.1 was released yesterday. I think this issue can be closed safely.
